themes/default/content/resources/walkthrough-of-building-container-based-aws-lambda-functions/index.md

```diff
@@ -0,0 +1,75 @@
+
+    So that's kind of, or access to any of them, rather, it's a set of constants, so that's how we bring that into there. And so once we have those, we just create the function. That's what this line is here on line 20. And so the function takes that image, a URI. So we have that image URI from above and this is an output of this particular image. And then we also give it the role that we created. So those things combined allow us to create the function that actually executes. And I'm going to go into how that function works in a second. After that, the rest of the code is actually fairly simple as well. We create a onObjectCreated. This is again a convenience in the Pulumi AWS library where we hook up the S3 bucket notifications to our function.
+
+    And so every time there's a new object created in that bucket. And in this particular bucket, we will the name of the handlers on new video. This could be any name. And then we give it the function that we want to execute whenever a new bucket object is created. And then we specifically filter for this MP4 suffix. And then finally, just to kind of, you know, for the sake of the demo, I also throw in an additional function that we create here. So again, we can create another onObjectCreated notification so that you can see it's actually the same thing, but here we can give it a new handler name in this case. Instead of giving it an existing function, I created a new function inline actually, and my inline function is simply just a TypeScript code I create here, and you can see the inline function is fairly simple.
+
+    All it's doing is providing some additional information about what's actually being processed. And again, we give it a policy. So that's, that's pretty much it here. And you can see this is, this is looking for that JPEG thumbnail that gets written once it gets written. And so, nothing too crazy here. So instead of a role, I just give it a policy because this actually creates a role on its own as well. So that's kind of how all that works. And so really the only code left look at is how do we build that container image and what is the code running within that container image? So if we go over here, we can look at our Docker file and you can see up here, I'm basing this image off of the image that AWS has published.
```

Comment:
```suggestion
    All it's doing is providing some additional information about what's actually being processed. And again, we give it a policy. So that's, that's pretty much it here. And you can see this is, this is looking for that JPEG thumbnail that gets written once it gets written. And so, nothing too crazy here. So instead of a role, I just give it a policy because this actually creates a role on its own as well. So that's kind of how all that works. And so really the only code left to look at is how do we build that container image and what is the code running within that container image? So if we go over here, we can look at our Docker file and you can see up here, I'm basing this image off of the image that AWS has published.
```